symengine/visitor.h

```diff
@@ -185,6 +185,39 @@ RCP<const Basic> coeff(const Basic &b, const Basic &x, const Basic &n);
 
 set_basic free_symbols(const Basic &b);
 
+class TransformVisitor : public BaseVisitor<TransformVisitor>
+{
+protected:
+    RCP<const Basic> result_;
+
+public:
+    TransformVisitor()
+    {
+    }
+
+    RCP<const Basic> apply(const Basic &x);
+    RCP<const Basic> apply(const RCP<const Basic> &x);
+
+    void bvisit(const Basic &x);
+    void bvisit(const Add &x);
+    void bvisit(const Mul &x);
+    void bvisit(const Pow &x);
+    void bvisit(const OneArgFunction &x);
+
+    template <class T>
+    void bvisit(const TwoArgBasic<T> &x)
```

Comment:
@isuruf, There were many linking errors while porting this templated member function to `.cpp` file. Should I leave it here?